theano/gof/type.py

```diff
@@ -710,8 +742,20 @@ def c_cleanup(self, name, sub):
         # free the data for us when released.
         return ""
 
+    def c_headers(self):
+        return self.headers
+
+    def c_header_dirs(self):
+        return self.header_dirs
+
+    def c_libraries(self):
+        return self.libraries
+
+    def c_lib_dirs(self):
+        return self.lib_dirs
+
     def c_code_cache_version(self):
-        return (2, self.ctype, self.freefunc)
+        return (3,)
```

Comment:
right! It is not mandatory, but I think the new parameter should be added in \__eq__ and \__hash__. Shoud we use \__props__ for this class instead?